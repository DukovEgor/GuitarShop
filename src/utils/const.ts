export enum HttpCode {
  BadRequest = 400,
  NotFound = 404,
}

export enum NameSpace {
  Data = 'data',
}

export enum APIRoute {
  Products = '/guitars?_limit=27',
  Product = '/guitars',
  Comments = '/comments',
  Coupons = '/coupons',
  Orders = '/orders',
}

export enum AppRoutes {
  Root = '/',
  Catalog = 'catalog',
  Product = 'product',
  Page = '/page_',
  DefaultPage = '/page_1',
  Characteristics = 'characteristics',
  Description = 'description',
}

export const PRUDUCTS_TO_SHOW = 9;
export const REVIEWS_TO_SHOW = 3;

export const INITIAL_PRODUCT = {
  id: 0,
  name: '',
  vendorCode: '',
  type: '',
  description: '',
  previewImg: '',
  stringCount: 0,
  rating: 0,
  price: 0,
};
