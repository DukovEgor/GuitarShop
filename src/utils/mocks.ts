import { IProduct } from '../interfaces/product';
import { faker } from '@faker-js/faker';
import { IComment } from '../interfaces/comment';
import { IReview } from '../interfaces/review';
import { InitialData, InitialProcess } from '../interfaces/initial-data';

export const makeFakeProduct = (): IProduct => ({
  id: Number(faker.random.numeric(2)),
  name: faker.commerce.product(),
  vendorCode: faker.random.alphaNumeric(),
  type: faker.commerce.department(),
  description: faker.commerce.productDescription(),
  previewImg: faker.internet.avatar(),
  stringCount: Number(faker.random.numeric(1)),
  rating: Number(faker.random.numeric(1)),
  price: Number(faker.commerce.price(100, 200, 0)),
  comments: [makeFakeComment()],
});

export const makeFakeComment = (): IComment => ({
  id: faker.random.numeric(2),
  userName: faker.name.firstName(),
  advantage: faker.commerce.productDescription(),
  disadvantage: faker.commerce.productDescription(),
  comment: faker.commerce.productDescription(),
  rating: Number(faker.random.numeric(1)),
  createAt: faker.date.future(),
  guitarId: Number(faker.random.numeric(2)),
});
export const makeFakeReview = (): IReview => ({
  guitarId: Number(faker.random.numeric(2)),
  userName: faker.commerce.product(),
  advantage: faker.commerce.productDescription(),
  disadvantage: faker.commerce.productDescription(),
  comment: faker.commerce.productDescription(),
  rating: Number(faker.random.numeric(1)),
});

export const INITIAL_DATA: InitialData = {
  products: Array.from({ length: 10 }, makeFakeProduct),
  product: makeFakeProduct(),
  productsCount: 10,
};

export const INITIAL_PROCESS: InitialProcess = {
  searchResult: Array.from({ length: 10 }, makeFakeProduct),
  sortedProducts: Array.from({ length: 10 }, makeFakeProduct),
};

export const mockComment = {
  id: '1211',
  userName: 'faker.name.firstName()',
  advantage: 'faker.commerce.productDescription()',
  disadvantage: 'faker.commerce.productDescription()',
  comment: 'faker.commerce.productDescription()',
  rating: 4,
  createAt: 'Thu Jun 02 2022 15:45:34 GMT+0300 (Moscow Standard Time)',
  guitarId: 2323,
};

export const mockComments = [mockComment];

export const mockProduct = {
  id: 1,
  name: 'Chester',
  vendorCode: '12312cfadf',
  type: 'electric',
  description: 'test',
  previewImg: 'https://assets.htmlacademy.ru/img/logo.v2.svg',
  stringCount: 6,
  rating: 4,
  price: 400,
  comments: [mockComment],
};

export const mockProducts = [mockProduct];
