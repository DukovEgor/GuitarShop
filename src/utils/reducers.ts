export enum ApiActions {
  Products = 'data/loadProducts',
  ProductData = 'data/loadProduct',
  NewReview = 'data/addComment',
  SearchRequest = 'process/setSearchResult',
  Coupon = 'cart/fetchCoupon',
}
