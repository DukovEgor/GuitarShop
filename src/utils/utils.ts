import { IComment } from '../interfaces/comment';

export const sortByDate: (a: IComment, b: IComment) => number = (a, b) => {
  const firstDate = new Date(a.createAt).getTime();
  const secondDate = new Date(b.createAt).getTime();
  return secondDate - firstDate;
};
export const capitalize = (s: string) => s && s[0].toUpperCase() + s.slice(1);

export default function toggleBodyLock(isModalOpened: boolean) {
  if (isModalOpened) {
    document.body.classList.add('lock');
  } else {
    document.body.classList.remove('lock');
  }
}
