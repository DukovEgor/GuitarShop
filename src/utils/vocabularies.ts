export const RatingVocabulary = {
  '0': 'Не задан',
  '1': 'Ужасно',
  '2': 'Плохо',
  '3': 'Средне',
  '4': 'Хорошо',
  '5': 'Отлично',
};
