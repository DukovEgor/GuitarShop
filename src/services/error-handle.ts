import request from 'axios';
import { toast } from 'react-toastify';
import { ErrorType } from '../types/error';
import { HttpCode } from '../utils/const';

export const errorHandle = (error: ErrorType): void => {
  if (!request.isAxiosError(error)) {
    throw error;
  }

  const { response } = error;

  if (response) {
    switch (response.status) {
      case HttpCode.BadRequest:
        toast.info(error.message);
        break;
      case HttpCode.NotFound:
        toast.info(error.message);
        break;
    }
  }
};
