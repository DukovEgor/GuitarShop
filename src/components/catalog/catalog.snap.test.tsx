import { configureMockStore } from '@jedmao/redux-mock-store';
import { Action, ThunkDispatch } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import { createAPI } from '../../services/api';
import { State } from '../../types/state';
import { mockProduct, mockProducts } from '../../utils/mocks';
import Catalog from './catalog';

const api = createAPI();
const middlewares = [thunk.withExtraArgument(api)];

const mockStore = configureMockStore<State, Action, ThunkDispatch<State, typeof api, Action>>(middlewares);
const store = mockStore({
  data: {
    products: mockProducts,
    product: mockProduct,
    isDataLoaded: true,
  },
});

describe('Component: Catalog', () => {
  test('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Catalog />
        </BrowserRouter>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
