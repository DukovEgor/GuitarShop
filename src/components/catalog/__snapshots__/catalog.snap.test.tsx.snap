// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Catalog should render correctly 1`] = `<div />`;
